### PR TITLE
Cache invalidation fixes

### PR DESCRIPTION
**Pull request by emilyviolet**
**Summary**

This patch removes COM and VCM caching in ComputeTempDeformChunk. The old scheme assumed that the molecular centre-of-mass positions were good to re-use for the whole timestep, but that's not true as the temperature compute will be used inside the integrator, so the COM positions and velocities will change in between invocations of ComputeTempDeformChunk::compute_scalar().

I fixed two instances where the computes were getting called unnecessarily:

1. FixNVTSllodChunk didn't correctly recognise that ComputeTempDeformChunk has a compatible bias, so it was unnecessarily calling compute_scalar() at the start of nh_v_temp(). This does not change the output at all.
 2. I re-ordered the calls to remove_bias() in nh_v_temp(), since ComputeTempDeformChunk::remove_bias() calls the COM compute every time it is invoked. This is unnecessary, and can be bypassed by calling remove_bias_all() once at the start of nh_v_temp() (thus working in thermal velocities for the whole thing, which is what we were doing anyway) and then calling restore_bias_all() at the end of nh_v_temp().

The cumulative effects of removing the caching changes the output of LAMMPS very slightly, although the qualitative behaviour is unchanged. It should be "correct" this way, although I would appreciate a second pair of eyes to make sure I haven't inadvertently broken anything in the process.

With these changes, the execution speed goes from 480.571 timesteps/s to 515.345 timesteps/s.

### CODE REVIEW

**Comment by emilyviolet** [comment]
Removing the calls to `com_compute` in `remove_bias` and friends seems to also fix the numerical MPI bug we were seeing before as well. That bug seemed to cause the temperature to have a large discontinuity when we do reneighbouring (I think), possibly due to the chunk IDs getting updated when atoms move between processors. This doesn't totally make sense to me, since I already had flags to force a rebuild of the chunk IDs if we've rebuilt the neighbour lists on this timestep, but removing the calls does let me run with as may MPI procs as I have cores.

**Review by ssande7 — APPROVED** [review]
This looks good! It seems to be standard practice to call `compute_scalar()` before `remove_bias_all()`, so I think assuming `com_compute()` has run is a good strategy.

Also nice that you can cover `vcm_thermal_compute()` and bias requirements during update with the single `remove_bias_all()`. Maybe worth keeping in mind to check whether the two calls of `remove_bias(..)` and `restore_bias(..)` are actually faster due to caching, but it's probably a minor difference if faster at all.

In case you missed them, did you want to clean up the TODO notes in the temp compute cpp file before merging this? I'll leave it up to you.

**Pull request merged**
